src/controllers/userController.ts

```diff
@@ -0,0 +1,59 @@
+import Database from "../database/db";
+import { Request, Response } from "express";
+import { UserAttributes } from "../database/models/user";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+import sendResponse from "../utils/response";
+
+dotenv.config();
+
+interface UserCreationAttributes extends Omit<UserAttributes, "id"> {}
+
+export const createUser = async (
+  req: Request<{}, {}, UserCreationAttributes>,
+  res: Response
+) => {
+  try {
+    const { name, email, phone, address, password } = req.body;
+
+    const userExist = await Database.User.findOne({
+      where: {
+        email,
+      },
+    });
+
+    if (userExist) {
+      return sendResponse<null>(
+        res,
+        400,
+        null,
+        "A user with this email already exists."
+      );
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    const user = Database.User.build({
+      name,
+      email,
+      phone,
+      address,
+      password: hashedPassword,
+    });
+
+    await user.save();
+
+    // const token = jwt.sign(
+    //   { id: user.id },
+    //   process.env.JWT_SECRET || "secret",
+    //   { expiresIn: "2h" }
+    // );
+    const tokenPayload = { email: user.email, password: user.password};
+    const token = jwt.sign(tokenPayload, process.env.ACCESS_TOKEN!);
```

Comment:
rename this environment variable to secret

Reply:
done